### PR TITLE
Feat/captcha

### PR DESCRIPTION
**Pull request by jona159**

### CODE REVIEW

**Review by YouQam — APPROVED** [review]
LGTM
`Note`: don't merge until approve/merge  [Feat/captcha #282 ](https://github.com/mint-o-badges/badgr-ui/pull/282/commits/0d92206773190b6c5cd739610000ea9466ee170f)

File: apps/badgeuser/api.py [hunk 1]
```diff
@@ -62,20 +86,58 @@ class BadgeUserDetail(BaseEntityDetailView):
...
+            _, email_domain = email.split("@", 1)
+            firstname = request.data.get("first_name")
+            lastname = request.data.get("last_name")
+
+            apiKey = getattr(settings, "ALTCHA_API_KEY")
+            endpoint = getattr(settings, "ALTCHA_SPAMFILTER_ENDPOINT")
+            payload = {
+                "text": [firstname, lastname],
+                # the following options seem to classify too much data as spam, i commented them out for now
+                # "email": email_domain,
```
[the remark this answers is not in the document]
Comment (jona159) [reply apps/badgeuser/api.py:113]: commented out email_domain as well

File: apps/badgeuser/api.py [hunk 2]
```diff
@@ -62,20 +86,58 @@ class BadgeUserDetail(BaseEntityDetailView):
...
+                # "expectedLanguages": ["en", "de"],
+            }
+            params = {"apiKey": apiKey}
+            headers = {
+                "Content-Type": "application/json",
+                "referer": getattr(settings, "HTTP_ORIGIN"),
+                # "referer": "https://api.openbadges.education/",
```
[the remark this answers is not in the document]
Comment (jona159) [reply apps/badgeuser/api.py:120]: deleted

File: apps/badgeuser/api.py [hunk 3]
```diff
@@ -62,20 +86,58 @@ class BadgeUserDetail(BaseEntityDetailView):
...
+            email = request.data.get("email")
+            # only send email domain to spamfilter API to protect users privacy
+            _, email_domain = email.split("@", 1)
+            firstname = request.data.get("first_name")
+            lastname = request.data.get("last_name")
+
+            apiKey = getattr(settings, "ALTCHA_API_KEY")
+            endpoint = getattr(settings, "ALTCHA_SPAMFILTER_ENDPOINT")
+            payload = {
+                "text": [firstname, lastname],
```
[the remark this answers is not in the document]
Comment (jona159) [reply apps/badgeuser/api.py]: I commented out the whole spamfilter for now, this topic will be discussed in a seperate meeting

File: apps/badgeuser/api.py [hunk 4]
```diff
@@ -62,20 +86,58 @@ class BadgeUserDetail(BaseEntityDetailView):
...
+                # "referer": "https://api.openbadges.education/",
+            }
+            response = requests.post(
+                endpoint, params=params, data=json.dumps(payload), headers=headers
+            )
+            if response.status_code == 200:
+                data = response.json()
+                classification = data["classification"]
+                if classification == "BAD":
+                    # TODO: show reasons why data was classified as spam
```
[the remark this answers is not in the document]
Comment (jona159) [reply apps/badgeuser/api.py]: need to wait for the results from abovementioned meeting

File: .docker/etc/settings_local.prod.py.example [hunk 5]
```diff
@@ -147,3 +147,7 @@ NOUNPROJECT_SECRET = ''
 
 AISKILLS_API_KEY = ''
 AISKILLS_ENDPOINT = ''
+
```
[the remark this answers is not in the document]
Comment (jona159) [reply .docker/etc/settings_local.prod.py.example:150]: Done

**Comment by timber-they** [comment]
@jona159 I approved the changes but now you need to resolve merge conflicts

**Pull request merged**